### PR TITLE
README.md: Refactor the README.md

### PR DESCRIPTION
**Pull request by Hoverbear**
This PR attempts to organize the README to be a bit more friendly to people interested in exploring TiKV.

# Motivation

We'd like to ensure that new users and contributors can easily get started on using TiKV. There are several prerequisites to building TiKV successfully, and it would be useful for new users to know common commands (such as `make`).

### CODE REVIEW

**Comment by breeswish** [comment]
@Hoverbear I'm curious what's in a full setup guide LOL, I though that the steps in master README is enough to set up just for TiKV.

**Comment by breeswish** [comment]
@Hoverbear Wow, nice catch! I have many related toolchain installed and I even didn't noticed this..

I would like a fully workable set up guide for TiKV in TiKV's README and provide a link to set up other components as supplementary info.

**Comment by Hoverbear** [comment]
@breeswish Great. I agree, let's move forward with that then. :)

**Comment by Hoverbear** [comment]
@breeswish Do you think it would be good to elaborate more or no?

**Comment by Hoverbear** [comment]
@breeswish Great idea. I'll reflect this.

**Comment by breeswish** [comment]
LGTM! @BusyJay PTAL

File: README.md [hunk 1]
```diff
@@ -2,76 +2,154 @@
...
+cd tidb-docker-compose
+docker-compose up -d
+```
+
+Shortly after you will be able to connect with `mysql -h 127.0.0.1 -P 4000 -u root` and view the cluster metrics at [http://localhost:3000/](http://localhost:3000/).
+
+
+## Setting Up a Development Workspace
+
+The TiKV codebase is primarily written in Rust, but has components written in C++ (RocksDB) and Go (for protobufs). In order to provide consistency and avoid opinion-based arguments, we make extensive use of linters and automated formatting tools. Additionally, due to Rust's youth we are currently utlizing nightly builds which provide access to many useful features.
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply README.md]: :woman_facepalming: Whoops.

File: README.md [hunk 2]
```diff
@@ -2,76 +2,154 @@
...
+
+Any pull requests made will automatically be tested by our CI systems, so making sure the full suite passes before creating your PR is not strictly required. **All merged PRs must have passing CI tests.**
+
+### Getting the Rest of the System Working
+
+In order to get other components ([TiDB](https://github.com/pingcap/tidb) and [PD](https://github.com/pingcap/pd) working we suggest you follow the [development guide](https://github.com/pingcap/docs/blob/master/dev-guide/development.md) as you will need to (at least) have `pd-server` working alongside `tikv-server` in order to do integration level testing.
+
+## Deploying To Production
+
+**Use Ansible?** The official [Ansible playbooks](https://github.com/pingcap/tidb-ansible) are the reccomended way of deploying a cluster of TiKV nodes alongside the other components of a cluster.
```
Comment (breeswish) [inline README.md]: There is a typo `reccomended`

Comment (breeswish) [reply README.md]: Also I prefer directly providing a link to `docs/op-guide/ansible-deployment` just like the other deployment options. That document provides a link to the source repo as well if user is interested.

**Comment by Hoverbear** [comment]
I have three LGTMs but no approvals. :rofl: 

Please when making a review comment that is a LGTM tap the "approval" option.

**Pull request merged**